src/ion/data_structures.rs

```diff
@@ -606,20 +606,37 @@ impl LiveRangeSet {
 
 #[derive(Clone, Debug)]
 pub struct InsertedMove {
-    pub pos_prio: PosWithPrio,
+    pub pos: ProgPoint,
     pub from_alloc: Allocation,
     pub to_alloc: Allocation,
     pub to_vreg: VReg,
 }
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u8)]
 pub enum InsertMovePrio {
     InEdgeMoves,
     Regular,
     MultiFixedRegInitial,
     MultiFixedRegSecondary,
     ReusedInput,
     OutEdgeMoves,
+
+    /// The number of priorities, used to determine the length of the Env::inserted_moves array.
```

Comment:
Let's leave a comment here too noting that any new priorities should be added to `InsertMovePrio::all()` below.